pkgs/by-name/se/segger-systemview/package.nix

```
@@ -0,0 +1,186 @@
+{ lib
+, stdenv
+, fetchurl
+, autoPatchelfHook
+, segger-jlink
+, config
+, acceptLicense ? config.segger-systemview.acceptLicense or false
+, fontconfig
+, xorg
+, makeDesktopItem
+, copyDesktopItems
+}:
+
+let
+  supported = {
+    x86_64-linux = {
+      name = "x86_64";
+      sha256 = "d3ec8d54ae0605a40d12d6eeea80967362ed3ed921395f154efee8d65d0c25ee";
+    };
+    i686-linux = {
+      name = "i386";
+      sha256 = "2292a6ed6c9b277e514ddbafb396e0e5a8ef8ca57ebd3236cf5ae29a52d2b5bc";
+    };
+  };
+
+  platform = supported.${stdenv.system} or (throw "unsupported platform ${stdenv.system}");
```

Maybe this should specify the target platform?

```suggestion
  platform = supported.${stdenv.targetPlatform.system} or (throw "unsupported platform ${stdenv.targetPlatform.system}");
```

---

I used the same pattern as in #255185 , but I think you are correct. Changed it to use `targetPlatform`.